hazelcast/src/main/java/com/hazelcast/internal/monitors/PerformanceLogFile.java

```diff
@@ -0,0 +1,322 @@
+            logger.finest("max rolling file count: " + maxRollingFileCount);
+        }
+    }
+
+    void render() {
+        try {
+            if (logFile == null) {
+                logFile = new File(format(pathname, index));
+                renderHead = true;
+            }
+
+            if (writer == null) {
+                writer = new BufferedWriter(new FileWriter(logFile, true));
+            }
+
+            if (renderHead) {
+                headRenderer.render(writer);
+                renderHead = false;
+            }
+
+            bodyRendered.render(writer);
+            writer.flush();
+            lastAccessed = logFile.lastModified();
+
+            if (logFile.length() > maxRollingFileSizeBytes) {
+                closeResource(writer);
+                writer = null;
+                logFile = null;
+                index++;
+                deleteOld();
+            }
+        } catch (IOException e) {
+            logIOError(e);
+            logFile = null;
+            closeResource(writer);
+            writer = null;
+        }
+    }
+
+    private void logIOError(IOException e) {
+        if (logger.isFinestEnabled()) {
+            logger.finest("PerformanceMonitor failed to output to file:"
+                    + logFile.getAbsolutePath() + " cause:" + e.getMessage(), e);
+        } else {
+            logger.warning("PerformanceMonitor failed to output to file:"
+                    + logFile.getAbsolutePath() + " cause:" + e.getMessage());
+        }
+    }
+
+    private void deleteOld() {
+        File file = new File(format(pathname, index - maxRollingFileCount));
+        file.delete();
+    }
+
+    public boolean forceRender() {
+        if (logFile == null) {
+            return false;
+        }
+
+        return logFile.lastModified() != lastAccessed;
+    }
+
+    private final class BodyRendered {
+
+        private final SimpleDateFormat sdf = new SimpleDateFormat("EEE, d MMM yyyy HH:mm:ss Z");
+        private final List<String> parameters = new ArrayList<String>();
+        private final List<Gauge> gauges = new ArrayList<Gauge>();
+        private final StringBuilder sb = new StringBuilder();
+        // Since the metricRegistry.modCount starts at 0, by making the lastModCount=-1, we are triggering
+        // the loading of the gauges on the first run.
+        private int lastModCount = -1;
+
+        private void render(BufferedWriter writer) throws IOException {
+            renderHeader(writer);
+            renderMetricsRegistry(writer);
+            writer.append("\n\n");
+            if (renderSlowOperation(writer)) {
+                writer.append("\n\n");
+            }
+        }
+
+        private void renderHeader(BufferedWriter writer) throws IOException {
+            writer.append("================[ Metrics ]============================================\n");
+            Calendar calendar = Calendar.getInstance();
```

Comment:
you construct a calendar instance only to print a timestamp, right?
```
Date now = new Date();
writer.append(sdf.format(now)).append('\n');
```

Check source code of `Calendar getInstance()`. I believe it's unnecessary in your case. (unless I misunderstood this part badly)